### PR TITLE
CI: Add examples to pytest

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
Include examples in pytest

### CODE REVIEW

File: python-avd/tests/models.py [hunk 1]
```diff
@@ -86,12 +87,24 @@ def __init__(self, name: str) -> None:
...
+            # Molecule paths
+            self.path = MOLECULE_PATH / name
+            inventory_path = self.path / "inventory/hosts.yml"
+
+        self._inventory = InventoryManager(loader=DataLoader(), sources=[inventory_path.as_posix()])
         self._vars = VariableManager(loader=DataLoader(), inventory=self._inventory)
         self.hosts = []
         for host in self._inventory.get_hosts():
+            if self.name.startswith("example-") and host.name in ["cvp", "cloudvision"]:
+                # Ignore CVP devices in examples without bloating the example with out test groups.
```
Comment (gmuloc) [inline python-avd/tests/models.py]: do you mean this?
```suggestion
                # Ignore CVP devices in examples without bloating the example without test groups.
```

File: python-avd/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from pathlib import Path
+from sys import path
+
+path.insert(0, str(Path(__file__).parents[1]))
```
Comment (gmuloc) [inline python-avd/tests/conftest.py:10]: why is this required?

**Review by gmuloc — COMMENTED** [review]
did you intend to push the .coverage?

File: python-avd/tests/conftest.py (hunk 2 above)
Comment (ClausHolbechArista) [reply python-avd/tests/conftest.py:10]: running pytest from IDE will not start it from the "python-avd" folder so to ensure you pick up the correct code.

Comment (ClausHolbechArista) [inline python-avd/tests/conftest.py:10]:
```suggestion
# Since the pyavd and schema_tools code it stored a layer deeper than the repo root,
# we need to add it to the path when running pytest from repo root.
# That is relevant when using pytest extensions in IDE (VSCode)
path.insert(0, str(Path(__file__).parents[1]))
```

File: python-avd/pyproject.toml [hunk 3]
```diff
@@ -157,5 +157,6 @@ extend = "../pyproject.toml"
 [tool.pytest.ini_options]
 markers = [
     "molecule_scenarios: Create test case fixtures 'molecule_host' or 'molecule_scenario' for the given molecule scenarios.",
+    "examples: Create test case fixtures 'molecule_host' or 'molecule_scenario' for the given example.",
```
Comment (ClausHolbechArista) [inline python-avd/pyproject.toml]:
```suggestion
```

File: python-avd/tests/conftest.py (hunk 2 above)
Comment (gmuloc) [reply python-avd/tests/conftest.py:10]: maybe a little comment on top of it? :)

Comment (gmuloc) [reply python-avd/tests/conftest.py:10]: AH  my diff was outdated... resolving

**Review by gmuloc — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
